Sources/Sentry/Profiling/SentryProfiledTracerConcurrency.mm

```diff
@@ -245,10 +279,24 @@
 )
 {
     if (isProfiling && [hub.getClient.options isContinuousProfilingEnabled]) {
-        SENTRY_CASSERT(hub.getClient.options.profiling.lifecycle == SentryProfileLifecycleTrace,
-            @"Expected trace profile lifecycle configuration.");
-        SENTRY_CASSERT([SentryContinuousProfiler isCurrentlyProfiling],
-            @"Expected a continuous profiler to be running for this configuration.");
+        if (hub.getClient.options.profiling.lifecycle != SentryProfileLifecycleTrace) {
+#    if SENTRY_TEST || SENTRY_TESTCI
+            SENTRY_CASSERT(NO, @"Expected trace profile v2 lifecycle configuration.");
+#    else
+            SENTRY_LOG_WARN(@"Expected trace profile v2 lifecycle configuration.");
+            return;
+#    endif // SENTRY_TEST || SENTRY_TESTCI
+        }
+
+        if (![SentryContinuousProfiler isCurrentlyProfiling]) {
+#    if SENTRY_TEST || SENTRY_TESTCI
+            SENTRY_CASSERT(
+                NO, @"Expected a continuous profiler to be running for this configuration.");
+#    else
+            SENTRY_LOG_WARN(
+                @"Expected a continuous profiler to be running for this configuration.");
+#    endif // SENTRY_TEST || SENTRY_TESTCI
```

Comment:
`l`: Maybe we could define a macro that calls `SENTRY_CASSERT` when `SENTRY_TEST || SENTRY_TESTCI` and calls `SENTRY_LOG_WARN` otherwise to reduce some duplication.